File: programmers/js_src/level00/120860.js
// 직사각형 넓이 구하기
function solution(dots) {
  var answer = 1;

  for (let i = 0; i <= 1; i++) {
    dots.sort((a, b) => a[i] - b[i]);
    answer *= dots[3][i] - dots[0][i];
  }

  return answer;
}

// 다른 풀이
// function solution(dots) {
//   let x = [];
//   let y = [];

//   for (let pos of dots) {
//     x.push(pos[0]);
//     y.push(pos[1]);
//   }

//   return (Math.max(...x) - Math.min(...x)) * (Math.max(...y) - Math.min(...y));
// }

File: programmers/java/level00/120860.java
// 직사각형 넓이 구하기

class Solution {
    public int solution(int[][] dots) {
        int maxX = Integer.MIN_VALUE;
        int minX = Integer.MAX_VALUE;
        int maxY = Integer.MIN_VALUE;
        int minY = Integer.MAX_VALUE;
        
        for (int[] dot : dots) {
            int x = dot[0];
            int y = dot[1];
            
            maxX = Math.max(maxX, x);
            minX = Math.min(minX, x);
            maxY = Math.max(maxY, y);
            minY = Math.min(minY, y);
        }
        
        return (maxX - minX) * (maxY - minY);
    }
}